nx/lib/nx/random.ex

```diff
@@ -551,6 +551,181 @@ defmodule Nx.Random do
     |> stop_grad()
   end
 
+  @doc """
+  Returns a sample from a multivariate normal distribution with given `mean` and `covariance` (matrix).
+
+  ## Options
+
+    * `:type` - a float type for the returned tensor
+
+    * `:shape` - shape of the returned tensor
+
+    * `:names` - the names of the returned tensor
+
+  ## Examples
+
+      iex> key = Nx.Random.key(12)
+      iex> {multivariate_normal, _new_key} = Nx.Random.multivariate_normal(key, Nx.tensor([0]), Nx.tensor([[1]]))
+      iex> multivariate_normal
+      #Nx.Tensor<
+        f32[1]
+        [0.735927939414978]
+      >
+
+      iex> key = Nx.Random.key(12)
+      iex> {multivariate_normal, _new_key} = Nx.Random.multivariate_normal(key, Nx.tensor([0, 0]), Nx.tensor([[1, 0], [0, 1]]))
+      iex> multivariate_normal
+      #Nx.Tensor<
+        f32[2]
+        [-1.3425945043563843, -0.40812060236930847]
+      >
+
+      iex> key = Nx.Random.key(12)
+      iex> {multivariate_normal, _new_key} = Nx.Random.multivariate_normal(key, Nx.tensor([0]), Nx.tensor([[1]]), shape: {3, 2}, type: :f16)
+      iex> multivariate_normal
+      #Nx.Tensor<
+        f16[3][2][1]
+        [
+          [
+            [0.326904296875],
+            [0.2176513671875]
+          ],
+          [
+            [0.316650390625],
+            [0.1109619140625]
+          ],
+          [
+            [0.53955078125],
+            [-0.8857421875]
+          ]
+        ]
+      >
+
+      iex> key = Nx.Random.key(12)
+      iex> {multivariate_normal, _new_key} = Nx.Random.multivariate_normal(key, Nx.tensor([0, 0]), Nx.tensor([[1, 0], [0, 1]]), shape: {3, 2})
+      iex> multivariate_normal
+      #Nx.Tensor<
+        f32[3][2][2]
+        [
+          [
+            [0.9891449809074402, 1.0795185565948486],
+            [-0.9467806220054626, 1.47813880443573]
+          ],
+          [
+            [2.2095863819122314, -1.529456377029419],
+            [-0.7933920621871948, 1.121195673942566]
+          ],
+          [
+            [0.10976295918226242, -0.9959557056427002],
+            [0.4754556119441986, 1.1413804292678833]
+          ]
+        ]
+      >
+  """
+  defn multivariate_normal(key, mean, covariance, opts \\ []) do
+    keys = split(key)
+    {multivariate_normal_split(keys[1], mean, covariance, opts), keys[0]}
+  end
+
+  @doc """
+  Same as `multivariate_normal/4` but assumes the key has already been split.
+  """
+  defn multivariate_normal_split(key, mean, covariance, opts \\ []) do
+    assert_key!(key)
+    opts = keyword!(opts, [:names, :type, :shape, method: :cholesky])
+    {type, shape} = validate_multivariate_normal_args(mean, covariance, opts)
+    mean = Nx.as_type(mean, type)
+    covariance = Nx.as_type(covariance, type)
+
+    a =
+      case opts[:method] do
+        :svd ->
+          {u, s, _} = Nx.LinAlg.svd(covariance)
+          u * Nx.sqrt(s)
+
+        :eigh ->
+          {s, u} = Nx.LinAlg.eigh(covariance)
+          u * Nx.sqrt(Nx.abs(s))
```

Comment:
Why take the sqrt of the eigenvalues here?

Reply:
Makes sense!
You can add a line to the docs that specifies that if covariance isn't positive semi-definite, it will be garbage in, garbage out, and get rid of the abs

Reply:
Done!